### ISSUE TITLE
Suggestion: submit to zsh upstream

### ISSUE THREAD

**Issue by aspiers** [kirelagin/prompt_kir#1]
Hi there, looks like some nice work here!  Funny that I wrote adam2 in 1999, and I'm only just discovering this successor over 20 years later.  I agree that my original code wasn't very readable.  Have you thought about converting to the prompt theme form and submitting it upstream to the zsh project?

Also I've noticed two minor issues - it leaves some unused columns at the right hand side, and also when pwd is long, shrinking the window causes a bunch of noise.  I think adam2 also suffers from this in certain circumstances.

**Comment by kirelagin** [comment]
Hi! Well, yes, I thought about it, but never actually got to doing this. The reason is that, as many other mini-projects, this one became _good enough_ for my daily use, but not quite good enough for upstreaming it anywhere. And, I guess, the main problem is that I’m not even sure that it is not worse than adam2 – maybe I forgot to implement some functionality or broke something without noticing. Maybe I should finally do that, if even you are saying so :).

**Comment by kirelagin** [comment]
> shrinking the window causes a bunch of noise

Yes, very annoying! But it was not annoying enough for me to sort this out, as I expect it will be quite hard :(.

> it leaves some unused columns at the right hand side

Tell me more! I have never seen it.
